polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -897,10 +898,19 @@ private void validateLocationForTableLike(
           // }
         },
         () -> {
-          if (location.startsWith("file:") || location.startsWith("http")) {
-            throw new ForbiddenException(
-                "Invalid location '%s' for identifier '%s': File locations are not allowed",
-                location, identifier);
+          List<String> allowedStorageTypes =
+              callContext
+                  .getPolarisCallContext()
+                  .getConfigurationStore()
+                  .getConfiguration(
+                      callContext.getPolarisCallContext(),
+                      PolarisConfiguration.SUPPORTED_CATALOG_STORAGE_TYPES);
+          if (!allowedStorageTypes.contains(StorageConfigInfo.StorageTypeEnum.FILE.name())) {
```

Comment:
I wonder if we should just have a list of "allowed prefixes" as well. May be easier than special casing this.

Reply:
Agreed, I think this can maybe wait until we have more prefixes in the future though. There are a lot of places in the code that will have to change.

Reply:
I wonder if we ever actually execute this code anymore. The storage configuration used to be optional, but I think it's always required now. The code treats it as optional in places, but the spec requires it for Catalog creation.